### DISCUSSION TITLE
Change nickname to a random number every 5 seconds

### DISCUSSION THREAD

**Discussion by whyisthesheep** [rigwild/discord-self-bot-console#47] [Show and tell]
I made a script to do what the title says (bear in mind idk JavaScript, so sorry if my code is bad 💀)
```js
{
    id()
    let loop = true
    while (loop) {
        min = Math.ceil(100000);
        max = Math.floor(999999);
        number = Math.floor(Math.random() * (max - min) + min);
        await api.changeNick(gid, number)
        await delay(5000)
    }
}
```
I have tested this and it works, feedback would be appreciated.

**Comment by rigwild** [comment]
Good job!

Here are some advice:

- Use `var` instead of `let` for the `loop` variable to be able to stop the loop by using `loop = false` (`var` defines the variables in the global scope, whereas `let` only in the current scope, which is inside `{ ... }` in this case).
- `Math.ceil` and `Math.floor` are used to round float numbers, so they are useless here (100000 is already an integer!)
- No need to redefine the `max` and `min` values in the loop

Here is the fixed version

```js
{
    id()

	const min = 100000
	const max = 999999

    var loop = true
    while (loop) {
        const number = Math.floor(Math.random() * (max - min) + min)
        await api.changeNick(gid, number)
        await delay(5000)
    }
}
```

**Reply by whyisthesheep** [reply]
Thank you for the feedback!
